luigi/worker.py

```diff
@@ -131,7 +131,11 @@ def __init__(self, task, worker_id, result_queue, status_reporter,
         self.worker_id = worker_id
         self.result_queue = result_queue
         self.status_reporter = status_reporter
-        self.worker_timeout = task.worker_timeout or worker_timeout
+        self.worker_timeout = (
+            # Don't override task.worker_timeout when it is 0, only when it is None. This allows to
```

Comment:
Thanks for the fix. I don't think we need this comment because the code itself is clear to read. It was a bug after all.

Reply:
If a comment like this one would have been there before, the bug probably wouldn't have been introduced in the first place. That was my motivation for adding it, to prevent the bug from reappearing in some future refactoring.

But I will remove the comment, if still think it is unnecessary.

Reply:
I think a better way is to cover this with a test case than explaining how a potential bug could be made.

Reply:
Ok, done. Added a test and removed the comment.